pyvista/core/grid.py

```diff
@@ -764,3 +764,41 @@ def gen_coords(i):
         grid.field_data.update(self.field_data)
         grid.copy_meta_from(self, deep=True)
         return grid
+
+    @property
+    def extent(self) -> tuple:
+        """Return or set the extent of the UniformGrid.
+
+        This is the same as :class:`Grid.dimensions` with the upper bounds
+        minus one.
+
+        Examples
+        --------
+        Create a ``UniformGrid`` and show its extent.
+
+        >>> import pyvista
+        >>> grid = pyvista.UniformGrid(dims=(10, 10, 10))
+        >>> grid.extent
+        (0, 9, 0, 9, 0, 9)
+
+        >>> grid.extent = (2, 5, 2, 5, 2, 5)
+        >>> grid.extent
+        (2, 5, 2, 5, 2, 5)
+
+        Note how this also modifies the grid bounds and dimensions. Since we
+        use default spacing of 1 here, the bounds match the extent exactly.
+
+        >>> grid.bounds
+        (2.0, 5.0, 2.0, 5.0, 2.0, 5.0)
+        >>> grid.dimensions
+        (4, 4, 4)
+
+        """
+        return self.GetExtent()
+
+    @extent.setter
+    def extent(self, extent: Sequence[int]):
+        """Set the extent of the UniformGrid."""
+        if len(extent) != 6:
+            raise ValueError('Extent must be a vector of 6 values.')
+        self.SetExtent(extent)
```

Comment:
It might be better style to choose a different name for the parameter than the name of the function itself.